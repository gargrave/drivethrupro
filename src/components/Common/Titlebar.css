:root {
  --titlebar-height: 60px;
}

.titlebar {
  background-color: #64777b;
  color: white;
  box-shadow: 0 1px 3px rgba(0, 0, 0, .12), 0 1px 2px rgba(0, 0, 0, .24);
  height: var(--titlebar-height);
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  padding-left: 20px;
  font-size: 28px;
  line-height: var(--titlebar-height);
}

.titlebar a {
  color: white;
}
