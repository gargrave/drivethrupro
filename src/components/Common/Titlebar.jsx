import React from 'react'
import { Link } from 'react-router-dom'
import { string } from 'prop-types'

import './Titlebar.css'

const Titlebar = props => {
  const { title, titleLink } = props
  return (
    <div className="titlebar">
      {!titleLink && <span className="title-unlinked">props.title</span>}
      {!!titleLink &&
        <Link to={titleLink} className="title-linked">
          {title}
        </Link>}
    </div>
  )
}

Titlebar.propTypes = {
  title: string.isRequired,
  titleLink: string
}

export default Titlebar
