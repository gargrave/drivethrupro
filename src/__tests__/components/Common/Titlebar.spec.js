import React from 'react'
import { shallow } from 'enzyme'

import Titlebar from '../../../components/Common/Titlebar'

describe('Titlebar', () => {
  let props
  let component

  beforeEach(() => {
    props = {
      title: 'AppTitle',
      titleLink: '/link'
    }

    component = shallow(<Titlebar {...props} />)
  })

  test('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  test('should render title as a link when a link is provided', () => {
    const linkedTitleCount = component.find('.title-linked').length
    const unlinkedTitleCount = component.find('.title-unlinked').length
    expect(linkedTitleCount).toEqual(1)
    expect(unlinkedTitleCount).toEqual(0)
  })

  test('should render title as unlinked text when no link is provided', () => {
    delete props.titleLink
    component = shallow(<Titlebar {...props} />)
    const linkedTitleCount = component.find('.title-linked').length
    const unlinkedTitleCount = component.find('.title-unlinked').length
    expect(linkedTitleCount).toEqual(0)
    expect(unlinkedTitleCount).toEqual(1)
  })
})
