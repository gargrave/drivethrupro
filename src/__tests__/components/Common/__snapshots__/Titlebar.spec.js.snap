// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Titlebar renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="titlebar"
>
    <Link
        className="title-linked"
        replace={false}
        to="/link"
    >
        AppTitle
    </Link>
</div>,
  "nodes": Array [
    <div
      className="titlebar"
>
      <Link
            className="title-linked"
            replace={false}
            to="/link"
      >
            AppTitle
      </Link>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Titlebar
        title="AppTitle"
        titleLink="/link"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "title": "AppTitle",
          "titleLink": "/link",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="titlebar"
>
          <Link
                    className="title-linked"
                    replace={false}
                    to="/link"
          >
                    AppTitle
          </Link>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="titlebar"
>
          <Link
                    className="title-linked"
                    replace={false}
                    to="/link"
          >
                    AppTitle
          </Link>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Titlebar
    title="AppTitle"
    titleLink="/link"
/>,
}
`;
